### PR TITLE
create dto, mapper and mapstruct added

### PR DESCRIPTION
**Pull request by OlexVoit**

### CODE REVIEW

File: src/main/java/mate/academy/bookshop/service/impl/BookServiceImpl.java [hunk 1]
```diff
@@ -1,28 +1,42 @@
...
 
     @Override
-    public Book save(Book book) {
-        return bookRepository.save(book);
+    public BookDto findById(Long id) {
+        Optional<Book> bookById = bookRepository.findBookById(id);
+
+        return bookMapper.toDto(bookById.orElseThrow(
+                () -> new EntityNotFoundException("Can't find a book by ID: " + id)
+        ));
```
Comment (TarasYurkovskyi) [inline src/main/java/mate/academy/bookshop/service/impl/BookServiceImpl.java]: You can use Optional.map for mapping it into DTO and inline byId variable. It would be prettier.

File: pom.xml [hunk 2]
```diff
@@ -74,9 +96,37 @@
...
+						</path>
+					</annotationProcessorPaths>
 				</configuration>
 			</plugin>
+
 		</plugins>
 	</build>
 
-</project>
+</project>
```
Comment (TarasYurkovskyi) [inline pom.xml:132]: End file with new line

**Review by TarasYurkovskyi — APPROVED** [review]
Good job)

File: src/main/resources/application.properties [hunk 3]
```diff
@@ -6,3 +6,4 @@ spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver
 spring.jpa.hibernate.ddl-auto=create-drop
 spring.jpa.show-sql=true
 
+
```
Comment (d1sam) [inline src/main/resources/application.properties]:
```suggestion
```

File: src/test/java/mate/academy/bookshop/BookshopApplicationTests.java [hunk 4]
```diff
@@ -6,8 +6,7 @@
 @SpringBootTest
 class BookshopApplicationTests {
 
-	@Test
-	void contextLoads() {
-	}
-
+    @Test
+    void contextLoads() {
+    }
```
Comment (d1sam) [inline src/test/java/mate/academy/bookshop/BookshopApplicationTests.java]: remove this unnecessary change for this hw

File: src/main/java/mate/academy/bookshop/controller/BookController.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+import mate.academy.bookshop.dto.CreateBookRequestDto;
+import mate.academy.bookshop.service.BookService;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
```
Comment (d1sam) [inline src/main/java/mate/academy/bookshop/controller/BookController.java]: better to set this inside application.properties

File: src/main/resources/application.properties [hunk 6]
```diff
@@ -6,3 +6,4 @@ spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver
 spring.jpa.hibernate.ddl-auto=create-drop
 spring.jpa.show-sql=true
```
Comment (d1sam) [inline src/main/resources/application.properties:7]:
```suggestion
spring.jpa.show-sql=true
spring.jpa.open-in-view=false
spring.jackson.deserialization.fail-on-unknown-properties=true
```

File: src/main/java/mate/academy/bookshop/service/impl/BookServiceImpl.java (hunk 1 above)
Comment (d1sam) [reply src/main/java/mate/academy/bookshop/service/impl/BookServiceImpl.java]: Agree with @TarasYurkovskyi , it can be simplified :)

File: src/main/java/mate/academy/bookshop/model/Book.java [hunk 7]
```diff
@@ -20,14 +21,12 @@ public class Book {
     private String title;
     @Column(nullable = false)
     private String author;
+    @ToString.Exclude
```
Comment (d1sam) [inline src/main/java/mate/academy/bookshop/model/Book.java]: why?

**Review by d1sam — CHANGES_REQUESTED** [review]
Good job, but some changes should be done!

File: pom.xml (hunk 2 above)
Comment (bhdnchui) [reply pom.xml:132]: Please, add empty line.

**Review by bhdnchui — CHANGES_REQUESTED** [review]
Good job!
Just one minor comment still exists.
